client/packages/programs/src/JsonForms/components/PeriodSearch.tsx

```diff
@@ -63,20 +63,31 @@ const UIComponent = (props: ControlProps) => {
     ? (data.pages[data.pages.length - 1]?.pageNumber ?? 0)
     : 0;
 
-  const onChange = async (period: PeriodFragment) => {
+  const onChange = async (period: PeriodFragment | null) => {
     setPeriod(period);
-
-    if (path === 'periodId') {
-      handleChange(path, period.id);
+    if (period === null) {
+      handleChange(path, undefined);
+      handleChange('before', undefined);
```

Comment:
Need additional if clause to handle clearing of input selection.

Previously inputs couldn't be cleared - and if they were, they wouldn't actually change the query params.